pom.xml

```diff
@@ -68,13 +70,39 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
-          <configuration>
-            <release>${javaVersion}</release>
-            <jdkToolchain>
-              <version>[11,)</version>
-            </jdkToolchain>
-          </configuration>
+          <executions>
+            <execution>
+              <!-- Overwrite default-compile configuration -->
+              <id>default-compile</id>
+              <configuration>
+                <release>${javaVersion}</release>
+                <jdkToolchain>
+                  <version>[11,)</version>
+                </jdkToolchain>
+              </configuration>
+            </execution>
+            <execution>
+              <!-- Overwrite default-testCompile configuration -->
+              <id>default-testCompile</id>
+              <configuration>
+                <release>${testJavaVersion}</release>
+                <jdkToolchain>
+                  <version>[${testJavaVersion},)</version>
+                </jdkToolchain>
+              </configuration>
+            </execution>
+          </executions>
         </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>3.0.0-M5</version>
+          <!-- Note: Ideally this would specify `jdkToolchain` to make sure 
+            a JDK >= ${testJavaVersion} is used, but unlike other plugins maven-surefire-plugin 
+            seems to fail when no Maven toolchains are configured locally -->
```

Comment:
The reason why this works for the compiler plugin is most likely a "quirk" (or bug?) because it simply [ignores any exception](https://github.com/apache/maven-compiler-plugin/blob/6335382f97aad2e69b9f9a0a4af3cf801cd29273/src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java#L1472-L1476) when the toolchain lookup fails.
Maybe we run into the same problem there as well once they fix that.